File: src/main/java/com/dev/social/service/user/PostService.java
package com.dev.social.service.user;

import com.dev.social.dto.request.user.PostRequest;
import com.dev.social.dto.response.PostResponseDTO;
import com.dev.social.entity.Post;

import java.io.IOException;
import java.util.List;

public interface PostService {
    void addPost(PostRequest request) throws IOException;

    List<PostResponseDTO> getAllPosts();

    List<PostResponseDTO> getFriendPost();

    List<PostResponseDTO> getPostsByUser(String postId);

    List<PostResponseDTO> getPostById(String postId);

    void deletePost(String id);

    void updatePost(Post post);
}

File: service/PostService.ts
import axiosInstance from "./AxiosService";

const addPost = async (formData: FormData) => {
    return await axiosInstance.post('/posts', formData);
}

const getAllPosts = async () => {
    return await axiosInstance.get('/posts');
}

const getPostCurrent = async () => {
    return await axiosInstance.get('/posts/user-posts');
}
const getPostById = async (id: String) => {
    return await axiosInstance.get(`/posts/${id}`);
}

const getPostByUserId = async (userId: String) => {
    return await axiosInstance.get(`/posts/user-posts/${userId}`);
}

const deletePostById = async (postId: string) => {
    return await axiosInstance.delete(`/posts/${postId}`);
}

export default {
    addPost,
    getAllPosts,
    deletePostById,
    getPostById,
    getPostCurrent,
    getPostByUserId
};